src/adapter/src/coord.rs

```diff
@@ -471,6 +472,10 @@ struct ConnMeta {
 
     /// Channel on which to send notices to a session.
     notice_tx: mpsc::UnboundedSender<AdapterNotice>,
+
+    /// The current role of the session. This is used for query cancellation to get information
+    /// about the role that initiated a query. It must be updated if a session's role changes.
+    pub(crate) current_role: RoleId,
```

Comment:
I looked through the PostgreSQL to see how they handle this. They actually don't use the current role or the session role. They use the authenticated role. The authenticated role is the role that the current session authenticated with at connection time and it never changes for the entire duration of the session.

I haven't added the authenticated role to Materialize yet unfortunately. However, you can remove the comment here about updating the role because it should never get updated. Can you also update the first part to indicate that it's the authenticated role and not the current role.

In case you're interested in the PG implementation, I've included some links below:

Here is where `pg_cancel_backend` is implemented, which mostly delegates to `pg_signal_backend`:
https://github.com/postgres/postgres/blob/a14e75eb0b6a73821e0d66c0d407372ec8376105/src/backend/storage/ipc/signalfuncs.c#L110-L136

Here is the beginning of `pg_signal_backend`, where the metadata for the other process is retrieved and the permission check is done:
https://github.com/postgres/postgres/blob/a14e75eb0b6a73821e0d66c0d407372ec8376105/src/backend/storage/ipc/signalfuncs.c#L30-L84

Here is when a session's role in `static PGPROC *allProcs` get's initialized: 
https://github.com/postgres/postgres/blob/a14e75eb0b6a73821e0d66c0d407372ec8376105/src/backend/utils/init/miscinit.c#L778-L780

Reply:
Whoa, awesome sleuthing, @jkosh44!